message.go

```diff
@@ -45,7 +45,15 @@ func (m *Message) encode(pe packetEncoder) error {
 	pe.putInt8(attributes)
 
 	if m.Version >= 1 {
-		pe.putInt64(m.Timestamp.UnixNano() / int64(time.Millisecond))
+		timestamp := int64(-1)
+
+		if m.Timestamp.After(time.Unix(0, -1)) {
```

Comment:
I inverted the condition to use ```!m.Timestamp.Before``` as you suggested. Let me know if there is anything else.